kotlinx-coroutines-core/common/src/Job.kt

```diff
@@ -252,9 +252,9 @@ public interface Job : CoroutineContext.Element {
      * suspending function is invoked or while it is suspended, this function
      * throws [CancellationException].
      *
-     * In particular, it means that a parent coroutine invoking `join` on a child coroutine that was started using
-     * `launch(coroutineContext) { ... }` builder throws [CancellationException] if the child
-     * had crashed, unless a non-standard [CoroutineExceptionHandler] is installed in the context.
+     * In particular, it means that a parent coroutine invoking `join` on a child coroutine throws
```

Comment:
I'd suggest removing the whole paragraph, now it seems to be redundant. 
Also, please fix the same phrasing for `cancelAndJoin`

Reply:
I think this explanation still makes sense as it is quite counter-intuitive for people that you can crash an attempt to `join` another coroutine.